sync/metrics.go

```diff
@@ -24,27 +36,96 @@ func newMetrics() (*metrics, error) {
 		return nil, err
 	}
 
+	syncLoopStarted, err := meter.Int64Counter("sync_loop_started", metric.WithDescription("sync loop started"))
+	if err != nil {
+		return nil, err
+	}
+
+	trustedPeersOutOfSync, err := meter.Int64Counter("tr_peers_out_of_sync", metric.WithDescription("trusted peers out of sync"))
+	if err != nil {
+		return nil, err
+	}
+
+	laggingHeadersStart, err := meter.Int64Counter("sync_lagging_hdr_start", metric.WithDescription("lagging header start"))
+	if err != nil {
+		return nil, err
+	}
+
+	subjectiveHead, err := meter.Int64ObservableGauge("sync_subjective_head", metric.WithDescription("subjective head height"))
+	if err != nil {
+		return nil, err
+	}
+
+	headerTimestamp, err := meter.Float64Histogram("sync_subjective_head_ts",
+		metric.WithDescription("subjective_head_timestamp"))
+	if err != nil {
+		return nil, err
+	}
+
 	m := &metrics{
-		totalSyncedGauge: totalSynced,
+		ctx:                   context.Background(),
```

Comment:
why always background?

Reply:
we should store the metric in any case. So, I decided to create a background cox

Reply:
let's use background everywhere then